src/tools/pbs_upgrade_job.c

```diff
@@ -194,224 +286,258 @@ check_job_file(int fd)
 {
 	off_t pos_saved;
 	off_t pos_new;
-	int ret = -1;
-	int len;
-
+	int ret_version = -1;
+	int length = -1;
+	jobfix_PRE19 old_jobfix_pre19;
 	/* Save our current position so we can comeback to it */
 	pos_saved = lseek(fd, 0, SEEK_CUR);
 
+	/*---------- For PBSPro 12 version ---------- */ 
 	/* The string "Job_Name" should appear at address 01160 (octal) for v12 */
 	pos_new = lseek(fd, 01160, SEEK_SET);
 	if (pos_new != 01160) {
 		fprintf(stderr, "File format not recognized.\n");
 		goto check_job_file_exit;
 	}
 	errno = 0;
-	len = read(fd, &buf, 8);
-	if (len < 0) {
+	length = read(fd, &buf, 8);
+	if (length < 0) {
 		fprintf(stderr, "Failed to read input file [%s]\n",
-			errno ? strerror(errno) : "No error");
+				errno ? strerror(errno) : "No error");
 		goto check_job_file_exit;
 	}
-	if (len != 8) {
+	if (length != 8) {
 		fprintf(stderr, "Format not recognized, no attributes.\n");
 		goto check_job_file_exit;
 	}
 	buf[8] = '\0';
 	if (strcmp(buf, ATTR_name) == 0) {
-		ret = 12;
+		ret_version = 12;
 		goto check_job_file_exit;
 	}
 
-	/* The string "Job_Name" should appear at address 01760 (octal) for v13 */
-	pos_new = lseek(fd, 01760, SEEK_SET);
-	if (pos_new != 01760) {
-		fprintf(stderr, "File format not recognized.\n");
-		goto check_job_file_exit;
-	}
-	errno = 0;
-	len = read(fd, &buf, 8);
-	if (len < 0) {
+	/*---------- For PBSPro >=13 or <=18 versions job structure ---------- */
+	length = -1;
+	pos_new = lseek(fd, pos_saved, SEEK_SET);
+	memset(&old_jobfix_pre19, 0, sizeof(old_jobfix_pre19));
+	length = read(fd, (char *)&old_jobfix_pre19, sizeof(old_jobfix_pre19));
```

Comment:
memset and read for the whole structure is not needed to just find out the version of the structure. We can only read the version and then decide accordingly.

Reply:
done!